modules/core/src/main/java/org/apache/synapse/mediators/elementary/Target.java

```diff
@@ -459,6 +466,18 @@ public void insertJson(MessageContext synCtx, Object sourceJsonElement, SynapseL
                 }
                 break;
             }
+            case EnrichMediator.VARIABLE:
+                if (action.equalsIgnoreCase(ACTION_REPLACE)) {
```

Comment:
If we get a requirement to enrich to variables in other mediator, this may lead to a conflict. WDYT?

Reply:
Since we are not giving priority to the Enrich mediator going forward, we don't have to consider this.